Source/DafnyLanguageServer/Handlers/DafnyCodeActionHandler.cs

```diff
@@ -193,8 +193,13 @@ public class DafnyCodeActionInput : IDafnyCodeActionInput {
   }
 
   public string Extract(Range range) {
+    if (!CodeLineToPos.ContainsKey(range.Start.Line)) {
+      return ""; // Out of range
+    }
     var startTokenPos = CodeLineToPos[range.Start.Line] + range.Start.Character;
-    var endTokenPos = CodeLineToPos[range.End.Line] + range.End.Character;
+    var endTokenPos =
+      !CodeLineToPos.ContainsKey(range.Start.Line) ? Code.Length - startTokenPos :
```

Comment:
How can `CodeLineToPos.ContainsKey(range.Start.Line) ` ever be false here if you already return on line 197 when it's false?